test-integration/server/session/cache/TransactionCacheIT.java

```diff
@@ -371,6 +372,19 @@ public void whenInsertingAndDeletingInferredAttribute_instanceIsTracked(){
         assertFalse(tx.cache().getInferredInstances().anyMatch(inst -> inst.equals(attribute)));
     }
 
+    @Test
+    public void whenInsertingAttribute_attributeIsCached(){
+        AttributeType<String> attributeType = tx.putAttributeType("resource", AttributeType.DataType.STRING);
+        String value = "banana";
+        Attribute attribute = AttributeTypeImpl.from(attributeType).create(value);
+        String index = Schema.generateAttributeIndex(attributeType.label(), value);
+
+        Attribute cachedAttribute = tx.cache().getAttributeCache().get(index);
+        assertNotNull(cachedAttribute);
+        assertEquals(attribute, cachedAttribute);
+        assertEquals(attribute, attributeType.attribute(value));
+    }
```

Comment:
to avoid memory leaks in the future can you add a trivial test to check the attribute gets removed from cache if deleted?